mods/ts/rules/structures.yaml

```diff
@@ -169,6 +169,29 @@ GAPILE:
 	Power:
 		Amount: -20
 
+GAPAVE:
+	Tooltip:
+		Name: Pavement
+	Valued:
+		Cost: 75
+	Buildable:
+		Queue: Defense
+		BuildPaletteOrder: 30
+		Prerequisites: barracks
+	Building:
+		Adjacent: 4
+		TerrainTypes: Clear, DirtRoad, Rough
+		BuildSounds: place2.aud
+		AllowInvalidPlacement: true
+		Footprint: xx xx
+		Dimensions: 2,2
+	KillsSelf:
+		RemoveInstead: true
+	LaysTerrain:
+		Template: 671
```

Comment:
TS LATs the concrete (smooths the edges, like we do with D2K spice) using 9 different templates, which means this simple approach isn't really going to work.  The simplest way I can see is to have a specific TSConcrete layer that accepts a bool (is concrete yes/no), and then picks the right template to use for each tile based on its neighbours.